### PR TITLE
prepare sync-charts script for syncing master branch

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
We often have the case where we want to install the master branch, just because it has some new features or fixes that are important in very special environments. This PR makes it so that the sync-charts script can handle syncing the master branch as well (another PR will then change the Prow postsubmit to make it run for every branch).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/hack/ci/ci-sync-charts.sh [hunk 1]
```diff
@@ -7,14 +7,24 @@ source $(dirname $0)/../lib.sh
...
+DASHBOARD_VERSION="$KUBERMATIC_VERSION"
+
 # find out which kubermatic-installer branch to synchronize to
-if grep -qE '^v[0-9]+\.[0-9]+\.[0-9]+.*' <<< "$PULL_BASE_REF"; then
+if [ "$PULL_BASE_REF" == "master" ]; then
+  INSTALLER_BRANCH="master"
+  # do not use the actual head hash to avoid spamming the installer
+  # repo for every new commit on master branch
+  KUBERMATIC_VERSION="latest"
+  DASHBOARD_VERSION="master"
```
Comment (xrstf) [inline api/hack/ci/ci-sync-charts.sh:26]: `latest` is the latest tagged release for the dashboard repo, whereas in Kubermatic `latest` is just the latest non-test build I think.

File: api/hack/ci/ci-sync-charts.sh [hunk 2]
```diff
@@ -7,14 +7,24 @@ source $(dirname $0)/../lib.sh
 cd $(dirname $0)/../../..
 git fetch --tags
 
-# we only synchronize charts for tags
-if ! git tag -l | grep -q "^$PULL_BASE_REF\$"; then
-  echo "Base ref $PULL_BASE_REF is not a tag! Exiting..."
+# we only synchronize charts for the master branch and tagged versions
+if [ "$PULL_BASE_REF" != "master" ] && ! git tag -l | grep -q "^$PULL_BASE_REF\$"; then
```
Comment (kdomanski) [inline api/hack/ci/ci-sync-charts.sh]: I think a few brackets would add readability

**Comment by kdomanski** [comment]
/approve

Looks roughly fine, but I'd like more eyes on this.

File: api/hack/ci/ci-sync-charts.sh (hunk 2 above)
Comment (xrstf) [reply api/hack/ci/ci-sync-charts.sh]: I had the second part in `[ ... ]`, but then bash didn't want to parse it anymore... :D I'm open to suggestions.

**Pull request merged**